### ISSUE TITLE
ZimWiki writer numbers all the ordered list items with 1

### ISSUE THREAD

**Issue by pawelszramowski** [jgm/pandoc#4962]
Zim Wiki itself uses incrementing numbers for list items. It will also correctly display incorrectly numbered lists and update them on save. But why not do it correctly? The incrementation code is already there, it's just commented out: https://github.com/jgm/pandoc/blob/2aa417a2e57f88f34f0f8d4c55d3e621f788216b/src/Text/Pandoc/Writers/ZimWiki.hs#L260

I stumbled upon this issue while investigating migration from WikidPad to CherryTree and converting from HTML to Zim in pandoc-2.3.1-windows-x86_64.

**Comment by jgm** [comment]
@alexivkin

**Comment by alexivkin** [comment]
I'll take a look at it

**Comment by pawelszramowski** [comment]
Zim Wiki uses continuous numbering according to my observations. It also alternates between numbers and letter with each nesting level what ZimWiki writer doesn't do.

Pandoc's output is correctly interpreted by Zim itself, but problems appear in cases like mine when it's used as an intermediate format. I agree that Zim importer in CherryTree has got some issues. However, I think pandoc's output should be consistent with output of native application for a given format.

**Comment by jgm** [comment]
I'd be happy to change it to consecutive numbering, but I'm concerned by @alexivkin's report that "it was causing issues for no benefit."  Do you remember what the issues were?

**Comment by alexivkin** [comment]
IIRC It was cosmetic - the number sequence was continuing even when numbering groups were terminated by blocks of text. It looked wrong in the resulting text. However Zim does not care. It renumbers all blocks correctly when loading the file.

So I opted to keep everything numbered as "1." instead of numbering consequently, but not stopping at text blocks.

**Comment by jgm** [comment]
Alex Ivkin <notifications@github.com> writes:

> IIRC It was cosmetic - the number sequence was continuing even when numbering groups were terminated by blocks of text. It looked wrong in the resulting text. However Zim does not care. It renumbers all blocks correctly when loading the file.
>
> So I opted to keep everything numbered as "1." instead of numbering consequently, but not stopping at text blocks.

Thanks. Presumably we could change the programming so
new lists always start with 1, right?

**Comment by alexivkin** [comment]
Yes, but my haskell skills and the build envrionment are not up to par, I'll be a while until I can get to it.

**Comment by jgm** [comment]
@alexivkin I see, you probably ran into problems because of the way you're using a state variable to hold the number.

Here's a more Haskelly way to do it.  Remove stItemNum from state.
Add a parameter for the item number to `orderedListItemToZimWiki`.
So,
``` haskell
orderedListItemToZimWiki :: PandocMonad m
                         => WriterOptions -> Int -> [Block] -> ZW m String
```
Then, instead of
```haskell
  contents <- mapM (orderedListItemToZimWiki opts) items
```
you can do
```haskell
  contents <- zipWithM (orderedListItemToZimWiki opts) [1..] items
```

**Comment by jgm** [comment]
Closed by 714e0eb83433cc319ac5fe3eaa570ae2c3c80fdc
